packages/studio-web/src/app/upload/upload.component.ts

```diff
@@ -77,24 +85,28 @@ export class UploadComponent implements OnInit {
         $localize`Whoops, something went wrong while recording!`
       );
     });
-    this.rasService.getLangs$().subscribe({
-      next: (langs: Array<SupportedLanguage>) => {
-        this.langs = langs.sort((a, b) => {
-          if (a.names['_'] < b.names['_']) return -1;
-          if (a.names['_'] > b.names['_']) return 1;
-          return 0;
-        });
-      },
-      error: (err) => this.reportRasError(err),
-    });
   }
 
-  async ngOnInit(): Promise<void> {
+  async ngOnInit() {
```

Comment:
Will the call to `getLangs$()` wait until after ssjs is fully initialized?

Since `getLangs$()` is used as a subscription without waiting, while we await on ssjs, I would have intuitively inverted the two blocks in this function, so they can happen in parallel.

Reply:
you're absolutely right!  also, I am going to change it to download the models as well, as I mentioned in #130

Reply:
This is a bit more complicated to do correctly, actually, though I will do it - the issue is that we want to correctly handle the error where SoundSwallower cannot be loaded in the same way we handle a failure to contact the API.  The right way to do this is, I think to make an `Observable` (or a `Subject`, whatever) for the initialization and do it in "parallel" with the `/langs` loading, then catch the errors for both of them.

Reply:
Automatically testing a SoundSwallower load failure is something I'm not sure how to do either - currently I temporarily replace `soundswallower.web.wasm` with an empty file and rebuild everything.

Reply:
this is done, perhaps not in the prettiest or most Angularly correct way, but in a way that avoids a less-than-typesafe Subject or a global error handler